### DISCUSSION TITLE
Executing MsSQL stored procedure from an Airflow Task always ends OK

### DISCUSSION THREAD

**Discussion by thegunner157** [apache/airflow#35569] [General]
Hi, I have a question regarding MSSQLOperator. I have used it to run SSIS packages and stored procedures. And no matter how the stored procedure ends (even with failure), task is marked as success. Do you have any ideas how can I fix this behaviour? Can it be set up in post_execution callback (and if so how)? Or maybe is this matter of SQL code which runs the stored procedure?

**Comment by phi-friday** [comment]
It's not the same problem, but it's what I used to solve some of the errors I encountered while using MSSQL.

1. use `pyodbc` instead of `pymssql`
2. writing a task directly instead of the `SQLExecuteQueryOperator`, calling connection on the created task and using a sqlalchemy session directly (i.e. not using `SQLExecuteQueryOperator`, just hook)

**Reply by thegunner157** [reply]
Thanks for reply but what if I wanted to use MS SQL Server connection type not odbc ? Can I still use hooks etc?
